File: 1523-countOddNumbersIntervalRange/count_odds.rs
fn count_odds(low: i32, high: i32) -> i32 {
    let parity = (low % 2) * (high % 2);
    (high - low + 1) / 2 + parity

    // (2..5) === (1..4)
    // (1..5) =/= (0..4)

    // odd * odd = odd (to check both are odd)
    // (low..high) ((high - low + 1)/ 2) + low % 2 == 0
}

File: 1523-countOddNumbersIntervalRange/countOdds.c
int countOdds(int low, int high) {
  // The number of odd numbers between low and high (inclusive)
  // 1 .. 5 => 3 (1, 3, 5):  (5 - 1)/2 + 1 = 3
  // 2 .. 5 => 2 (3, 5):  (5 - 2)/2 + 1 = 2
  // 2 .. 4 => 1 (3):  (4 - 2)/2 + 0 = 1
  //
  // odd * odd = odd
  // odd * even = even
  // even * even = even
  int parity = (low % 2) || (high % 2);
  return (high - low) / 2 + parity;
}
